File: app/schemas/price.py
from __future__ import annotations
from pydantic import BaseModel, ConfigDict
from typing import Optional
from datetime import datetime, date
from decimal import Decimal


class DrugPriceBase(BaseModel):
    brand_id: int
    acquisition_price: Optional[Decimal] = None
    selling_price: Optional[Decimal] = None
    effective_date: date
    currency: str = "USD"


class DrugPriceCreate(DrugPriceBase):
    pass


class DrugPriceUpdate(BaseModel):
    acquisition_price: Optional[Decimal] = None
    selling_price: Optional[Decimal] = None
    effective_date: Optional[date] = None
    currency: Optional[str] = None


class DrugPriceInDB(DrugPriceBase):
    price_id: int
    created_at: datetime
    
    model_config = ConfigDict(from_attributes=True)


class DrugPriceResponse(DrugPriceInDB):
    pass


class DrugPriceWithBrand(DrugPriceResponse):
    brand_name: Optional["BrandNameResponse"] = None

File: frontend/src/types/price.ts
export interface DrugPrice {
  price_id: number;
  brand_id: number;
  acquisition_price?: number;
  selling_price?: number;
  effective_date: string;
  currency: string;
  created_at?: string;
}

export interface DrugPriceWithBrand extends DrugPrice {
  brand_name?: {
    brand_id: number;
    brand_name: string;
  };
}

export interface DrugPriceCreate {
  brand_id: number;
  acquisition_price?: number;
  selling_price?: number;
  effective_date: string;
  currency?: string;
}

export interface DrugPriceUpdate {
  acquisition_price?: number;
  selling_price?: number;
  effective_date?: string;
  currency?: string;
}
